src/Aspire.Hosting.Azure/AzureResourceExtensions.cs

```diff
@@ -15,6 +16,56 @@ namespace Aspire.Hosting;
 /// </summary>
 public static class AzureResourceExtensions
 {
+    public static IResourceBuilder<AzureBicepResource> AddBicepTemplate(this IDistributedApplicationBuilder builder, string name, string bicepFile)
+    {
+        var path = Path.GetFullPath(Path.Combine(builder.AppHostDirectory, bicepFile));
+        var resource = new AzureBicepResource(name, path);
+        return builder.AddResource(resource)
+                      .WithManifestPublishingCallback(context => WriteBicepResourceToManifest(context, resource));
+    }
+
+    public static IResourceBuilder<AzureBicepResource> WithParameter(this IResourceBuilder<AzureBicepResource> builder, string name, string value)
+    {
+        builder.Resource.Inputs[name] = value;
+        return builder;
+    }
+
+    public static IResourceBuilder<AzureBicepResource> WithParameter(this IResourceBuilder<AzureBicepResource> builder, string name, IResourceBuilder<ParameterResource> value)
+    {
+        builder.Resource.Inputs[name] = value;
+        return builder;
+    }
+
+    public static IResourceBuilder<AzureBicepResource> WithParameter(this IResourceBuilder<AzureBicepResource> builder, string name, IResourceBuilder<IResourceWithConnectionString> value)
```

Comment:
Should this be a special WithReference for consistency?

Reply:
With isn’t a reference. I’ll call it AddParameter